example/parsec/src/parsec.c

```diff
@@ -96,6 +96,62 @@ impl_ParsecLibrary(String);
 #define parseTest(p, input)                                              \
   GENERIC_PARSECRUNNER(String, p).parseTest(p, input)
 
+#define parse(p, state)                                                  \
+  GENERIC_PARSECRUNNER(String, p).runParsec(p, state)
+
+// -----------------------------------------------------------------------
+#define DO()                                                             \
+  g_bind((_s0_, _cok_, _cerr_, _eok_, _eerr_), *args);                   \
+  __auto_type _s_ = _s0_;
+
+#define SCAN(...) CAT(SCAN, VARIADIC_SIZE(__VA_ARGS__))(__VA_ARGS__)
+#define SCAN1(_p_)                                                       \
+  __auto_type TMPID = parse(_p_, _s_);                                   \
+  if (!TMPID.result.success) {                                           \
+    __auto_type _err_ = (TMPID.consumed ? _cerr_ : _eerr_);              \
+    return fn_apply(_err_, TMPID.result.err, TMPID.state);               \
+  }                                                                      \
+  _s_ = TMPID.state;
+#define SCAN2(_p_, _x_)                                                  \
+  SCAN1(_p_);                                                            \
+  __auto_type _x_ = TMPID.result.ok;
+
+#define RETURN(_x_)                                                      \
+  do {                                                                   \
+    __auto_type _ok_ = (_s0_.offset < _s_.offset ? _cok_ : _eok_);       \
+    return fn_apply(_ok_, _x_, _s_, (Hints(Token(S))){0});               \
+  } while (0)
+
+// -----------------------------------------------------------------------
+fn(identifierImpl, UnParserArgs(S, String)) {
+  __auto_type c = trait(ParsecChar(S));
+  __auto_type C = trait(ParsecChoice(S, char));
+  __auto_type m = trait(ParsecRepeat(S, char));
+  __auto_type identStart = C.either(c.char1('_'), c.letter());
+  __auto_type identLetter = m.many(C.either(c.char1('_'), c.alphaNum()));
+  __auto_type spaces = m.many(c.whitespace());
+
+  DO() {
+    SCAN(identStart, x);
+    SCAN(identLetter, xs);
+    SCAN(spaces);
+    size_t len = 1 + xs.length;
+    char* cs = mem_malloc(len + 1);
+    cs[0] = x;
+    memcpy(cs + 1, xs.data, xs.length);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/manual/mori0091/cparsec3/pullRequest?prid=5129640)